src/fu-util.c

```diff
@@ -1973,17 +1973,19 @@ fu_util_update_device_with_release (FuUtilPrivate *priv,
 			return FALSE;
 
 		/* local and directory remotes have the firmware already */
-		if (fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_LOCAL ||
-		    fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_DIRECTORY) {
+		if (fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_LOCAL) {
 			const gchar *fn_cache = fwupd_remote_get_filename_cache (remote);
 			g_autofree gchar *path = g_path_get_dirname (fn_cache);
 
-			/* install with flags chosen by the user */
 			fn = g_build_filename (path, uri_tmp, NULL);
+		} else if (fwupd_remote_get_kind (remote) == FWUPD_REMOTE_KIND_DIRECTORY) {
+			fn = g_strdup (uri_tmp + 7);
+		}
+		/* install with flags chosen by the user */
+		if (fn != NULL)
```

Comment:
why would fn be NULL? Isn't that an error we need to handle?

Reply:
Ohh, ignore me, I see now. I think adding back the braces would make this clearer.

Reply:
Added braces